contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -0,0 +1,319 @@
+        if (_underlying == address(0))
+            revert StrategyUnderlyingCannotBe0Address();
+        if (!_vault.doesContractImplementInterface(type(IVault).interfaceId))
+            revert StrategyNotIVault();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(MANAGER_ROLE, _vault);
+
+        vault = _vault;
+        underlying = IERC20(_underlying);
+        stabilityPool = IStabilityPool(_stabilityPool);
+        lqty = IERC20(_lqty);
+
+        if (!underlying.approve(_stabilityPool, type(uint256).max)) {
+            revert TokenApprovalFailed(_underlying);
+        }
+    }
+
+    /**
+     * Transfers administrator rights for the Strategy to another account,
+     * revoking current admin roles and setting up the roles for the new admin.
+     *
+     * @notice Can only be called by the account with the ADMIN role.
+     *
+     * @param _newAdmin The new Strategy admin account.
+     */
+    function transferAdminRights(address _newAdmin) external onlyAdmin {
+        if (_newAdmin == address(0x0)) revert StrategyAdminCannotBe0Address();
+
+        if (_newAdmin == msg.sender)
+            revert StrategyCannotTransferAdminRightsToSelf();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newAdmin);
+        _setupRole(MANAGER_ROLE, _newAdmin);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _revokeRole(MANAGER_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Returns true since strategy is synchronous.
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return true;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return investedAssets() > 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice ETH & LQTY rewards of the strategy waiting to be claimed in the liquity stability pool are not included
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return stabilityPool.getCompoundedLUSDDeposit(address(this));
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        stabilityPool.provideToSP(balance, address(0));
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice will also claim unclaimed LQTY & ETH gains
+    /// @notice when amount > total deposited, all available funds will be withdrawn
+    function withdrawToVault(uint256 amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (amount == 0) revert StrategyAmountZero();
+        if (amount > investedAssets()) revert StrategyNotEnoughShares();
```

Comment:
Do we need the first check against 0? I feel the second check should be enough because the worst-case scenario investedAssets() is 0, which means that `0 > 0` is false, right?

Reply:
makes sense, will fix that

Reply:
does it? xD if both are 0, then it should revert xd, but if you did `amount >= investedAssets`?

Reply:
I dont think the strategy should revert if amount == investedAssets.

Reply:
But about the first statement , " amount == 0", you make a good point here. I checked the Vault code, and there are two methods where withdrawToVault is being called by the Vault. And I think there wont be any scenario when withdrawToVault will be called by the vault with 0 amount. @gabrielpoca can you please confirm?

Reply:
I agree that the second check alone should be enough, as it currently is `amount > investedAssets()`. Also, it would save us some gas to have one check only.

Reply:
Maybe we could do it like this, remove the first check and change the second one to `(amount > investedAssets() && amount != 0)`. The second part `amount != 0` will not be executed if the first part is false, and we will not waste gas by having two separate checks performed on each call as it is currently. @gabrielpoca @dxganta wdyt?

Reply:
> does it? xD if both are 0, then it should revert xd, but if you did `amount >= investedAssets`?

Looks like we need both checks if we should revert when "amount = 0" :slightly_smiling_face:

Reply:
Even if the vault never calls when it is 0, that might change. This looks good as it is 👍